### PR TITLE
Fix: out of sync playhead countdown

### PR DESCRIPTION
**Pull request by thomasslee97**
This PR has been opened by SuperFly.tv on behalf of EVS Broadcast Equipment.

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix.

**What is the current behavior?** (You can also link to an open issue here)
When taking from one Part to another, the timing data for the previous Part lingers in the context provided by the `RundownTimingProvider` and is provided to the `CurrentPartRemaining` component. This results in the coundown continuing to display the countdown for the previous Part.

**What is the new behavior (if this is a feature change)?**
To fix this issue, two measures have been taken:
1. The `RundownTimingProvider` now reports what the `currentPartInstanceId` was when it performed the timing calculation as part of the `RundownTimingContext` object that it provides. This allows the `CurrentPartRemaining` component to not render the countdown timer if the timing data that it has received is stale. This does result in the timer being briefly hidden if up-to-date timing data does not arrive quickly.
2. To that end, the `RundownTimingProvider` has been updated to force all of the timing events (LR, HR, Synced) to be emitted whenever the current or next part instances change. This results in a much shorter period of time where there is stale data, and reduces the number of frames for which the coundown is hidden while it catches up with reality. I would like to stress that it is a matter of frames - during a screen capture I was unable to see any time where the coundown ended up hidden even when stepping through frame-by-frame. This is more of a safety guard to prevent bad data from being presented to the user.

**Other information**:
A change that has been made here is that the `syncedEventTimeNow` value in `RundownTimingProvider` has been removed so that `calmedDownNow` is used everywhere. This is because there was a disagreement between `syncedEventTimeNow` and `calmedDownNow` as to which second we were in, which caused a coundown of: `00:10`, `00:11`, `00:10`, `00:09`, for example. I am unsure if this is the right thing to do.

**Status**

- [X] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [X] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/client/ui/RundownView/RundownTiming/CurrentPartRemaining.tsx [hunk 1]
```diff
@@ -29,6 +29,8 @@ export const CurrentPartRemaining = withTiming<IPartRemainingProps, {}>({
 })(
 	class CurrentPartRemaining extends React.Component<WithTiming<IPartRemainingProps>> {
 		render(): JSX.Element {
+			if (!this.props.timingDurations || !this.props.timingDurations.currentTime) return <div></div>
+			if (this.props.timingDurations.currentPartInstanceId !== this.props.currentPartInstanceId) return <div></div>
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply meteor/client/ui/RundownView/RundownTiming/CurrentPartRemaining.tsx]: Won't the same be achievable without returning null if an empty fragment is returned?

File: meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx [hunk 2]
```diff
@@ -207,12 +207,11 @@ export const RundownTimingProvider = withTracker<
 				this.dispatchLREvent(calmedDownNow)
 			}
 
-			const syncedEventTimeNow = Math.floor(now / 1000) * 1000
```
Comment (ianshade) [inline meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx:210]: I believe `syncedEventTimeNow` makes other calculations rounded perfectly to the whole second represented by the synced event. Without it some values may start jumping unexpectedly e.g. you'll get timers that count like 00:01, 00:02, 00:02, 00:04

Comment (ianshade) [reply meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx:210]: Additionally I believe this made the synced timers tick as closely to the system clock as possible, which I don't think is the case anymore.

Comment (thomasslee97) [reply meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx:210]: Curiously, that sort of sequence of counting was what I was seeing before removing `syncedEventTimeNow`

Comment (ianshade) [reply meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx:210]: Was that happening before you made any changes attempting to fix this bug?

**Pull request merged**